src/main/java/seedu/address/commons/util/LoanSlipUtil.java

```diff
@@ -0,0 +1,217 @@
+     * @return a {@code Document} object representing the file.
+     * @throws IOException if there are errors in creating the new file.
+     */
+    private static Document createDocument(String docName) throws IOException, LoanSlipException {
+        if (!isMounted) {
+            throw new LoanSlipException("No Loan slip mounted yet");
+        }
+        String finalDest = DEST + docName + PDF_EXTENSION;
+        File file = new File(finalDest);
+        file.getParentFile().mkdirs();
+        PdfWriter writer = new PdfWriter(finalDest);
+        //Initialize PDF document
+        PdfDocument pdf = new PdfDocument(writer);
+        // Initialize document
+        currentFile = file;
+        return new Document(pdf);
+    }
+
+    /**
+     * Populates the pdf file with information, creating the loan slip.
+     *
+     * @param doc {@code LoanSlipDocument} object to be populated with data.
+     */
+    private static void generateLiberryLoanSlip(LoanSlipDocument doc) throws LoanSlipException {
+        if (!isMounted) {
+            throw new LoanSlipException("No Loan slip mounted yet");
+        }
+        //write logo
+        doc.writeLogo();
+        doc.writeLine();
+        //Add paragraph to the document
+        doc.writeHeader(currentBorrower.getName().toString());
+        doc.writeLeftParagraph(currentBorrower.getBorrowerId().toString());
+        doc.writeLine();
+        doc.writeMidHeader("Books borrowed");
+        //populate table
+        String[] headerRow = new String[]{"S/N", "Book", "Due By"};
+        doc.writeRow(headerRow);
+        doc.writeRow(createBookRow());
+        //add table to document
+        doc.submitTable();
+        doc.writeLine();
+        doc.writeCentralisedParagraph(BYE_MESSAGE);
+        //close document
+        doc.closeDoc();
+        isGenerated = true;
+    }
+
+    /**
+     * Creates an array using the details from the mounted loan slip.
+     *
+     * @return an array of string representing a row of the table.
+     */
+    private static String[] createBookRow() throws LoanSlipException {
+        if (!isMounted) {
+            throw new LoanSlipException("No Loan slip mounted yet");
+        }
+        String[] currentBookDetails = new String[3];
+        currentBookDetails[0] = currentBook.getSerialNumber().toString();
+        currentBookDetails[1] = currentBook.getTitle().toString();
+        currentBookDetails[2] = formatDate(currentLoan.getDueDate().toString());
+        return currentBookDetails;
+    }
+
+    /**
+     * Formats the date to be printed on the loan slip.
+     *
+     * @param date date to be formatted.
+     * @return a formatted date as a {@code String}
+     */
+    private static String formatDate(String date) {
+        String[] pieces = date.split("-");
+        String result = "";
+        for (String s : pieces) {
+            result = s + "-" + result;
+        }
+        return result.substring(0, result.length() - 1);
+    }
```

Comment:
This method can be inside DateUtil actually.

Reply:
Makes sense! will transfer it!